### PR TITLE
Enable CI

### PR DESCRIPTION
**Pull request by dns2utf8**
As requested by @ubruhin

### CODE REVIEW

**Comment by dns2utf8** [comment]
@dbrgn can you enable https://travis-ci.org/coredump-ch/rgb-led-firmware

**Comment by dbrgn** [comment]
Done :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+language: c
+before_install:
+  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_1.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :1 -ac -screen 0 1280x1024x16"
+  - sleep 3
+  - export DISPLAY=:1.0
+  - wget http://downloads.arduino.cc/arduino-1.6.5-linux64.tar.xz
```
Comment (rnestler) [inline .travis.yml]: Why such an old Arduino version?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - tar xf arduino-1.6.5-linux64.tar.xz
+  - sudo mv arduino-1.6.5 /usr/local/share/arduino
+  - sudo ln -s /usr/local/share/arduino/arduino /usr/local/bin/arduino
+install:
+  - ln -s $PWD /usr/local/share/arduino/libraries/Adafruit_FONA
+  - arduino --install-library "Adafruit SleepyDog Library,Adafruit MQTT Library"
+script:
+  - arduino --verify --board arduino:avr:uno $PWD/examples/FONAtest/FONAtest.ino
```
Comment (rnestler) [inline .travis.yml]: This is the wrong board. We use the Arduino nano.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+language: c
+before_install:
+  - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_1.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :1 -ac -screen 0 1280x1024x16"
+  - sleep 3
```
Comment (rnestler) [inline .travis.yml]: Why do we need an X-Server? We should use arduino-builder, which is a CLI app.

**Review by rnestler — CHANGES_REQUESTED** [review]
What should this do exactly? It seems to build some examples and not our firmware.
Also take a look at [arduino-builder](https://github.com/arduino/arduino-builder). It's included with the default Arduino installation.

File: .travis.yml (hunk 2 above)
Comment (dbrgn) [reply .travis.yml]: Doesn't really matter for the build process itself though.

Comment (rnestler) [reply .travis.yml]: It does if we start using board specific stuff.

**Review by rnestler — APPROVED** [review]

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+language: c
```
Comment (ubruhin) [inline .travis.yml]: Why "c" and not "cpp"?

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+env:
+  - ARDUINO=1.6.12
+install:
+  - echo $PWD
```
Comment (ubruhin) [inline .travis.yml]: What is this for?

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  - wget http://downloads.arduino.cc/arduino-$ARDUINO-linux64.tar.xz
+  - tar xf arduino-$ARDUINO-linux64.tar.xz
+script:
+  - echo $PWD
```
Comment (ubruhin) [inline .travis.yml]: What is this for?

File: .travis.yml (hunk 4 above)
Comment (rnestler) [reply .travis.yml]: Why anything at all. All that `language: c` does it provides a build environement for C and some default test scripts.

File: .travis.yml (hunk 5 above)
Comment (rnestler) [reply .travis.yml]: Just for debugging the build script :wink: wasn't sure about the path were the steps get executed.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+env:
+  - ARDUINO=1.6.12
+before_script:
+  - wget http://downloads.arduino.cc/arduino-$ARDUINO-linux64.tar.xz
+  - tar xf arduino-$ARDUINO-linux64.tar.xz
+script:
+  - arduino-$ARDUINO/arduino-builder -hardware arduino-$ARDUINO/hardware -tools arduino-$ARDUINO/tools-builder -tools arduino-$ARDUINO/hardware/tools/avr -fqbn arduino:avr:nano:cpu=atmega328 -warnings all -verbose src/src.ino
```
Comment (ubruhin) [inline .travis.yml]: Too long line, add line breaks to improve readability.

**Review by ubruhin — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+notifications:
+  email:
+    on_success: change
```
Comment (dns2utf8) [inline .travis.yml]: Can we reduce this to never?

Comment (rnestler) [reply .travis.yml]: I don't see why not, I don't use the email notifications anyway. Can I just remove the whole notifications section?

Comment (dns2utf8) [reply .travis.yml]: No, the default is set to spam :wink:

**Review by dbrgn — APPROVED** [review]

**Pull request merged**